tasks/post_to_database/post_to_database.py

```diff
@@ -0,0 +1,249 @@
+        file_parameters.append({'job_id': job_id, 'granule_id': granule_id, 'filename': file['filename'],
+                                'key_path': file['key_path'], 'restore_destination': file['restore_destination'],
+                                'status_id': file['status_id'], 'error_message': file.get('error_message', None),
+                                'request_time': file['request_time'], 'last_update': file['last_update'],
+                                'completion_time': file.get('completion_time', None)})
+
+    if found_pending:
+        # Most jobs will be this. Some files are still pending.
+        job_status = OrcaStatus.PENDING
+        job_completion_time = None
+    else:
+        # All files failed during recovery request.
+        job_status = OrcaStatus.FAILED
+        job_completion_time = job_completion_time.isoformat().__str__()
+
+    engine = shared_db.get_user_connection(db_connect_info)
+    with engine.begin() as connection:
+        connection.execute(create_job_sql(),
+                           [{'job_id': job_id, 'granule_id': granule_id, 'status_id': job_status.value,
+                             'request_time': request_time,
+                             'completion_time': job_completion_time,
+                             'archive_destination': archive_destination}])
+        connection.execute(create_file_sql(), file_parameters)
+
+
+def update_status_for_file(job_id: str,
+                           granule_id: str,
+                           filename: str,
+                           last_update: str,
+                           completion_time: Optional[str],
+                           status_id: int,
+                           error_message: Optional[str],
+                           db_connect_info: Dict) -> None:
+    """
+    Updates a given file's status entry, modifying the job if all files for that job have advanced in status.
+
+    Args:
+        job_id: The unique identifier used for tracking requests.
+        granule_id: The id of the granule being restored.
+        filename: The name of the file being copied.
+        last_update: The time this status update occurred, in UTC iso-format.
+        completion_time: The completion time, in UTC iso-format.
+        status_id: Defines the status id used in the ORCA Recovery database.
+        error_message: message displayed on error.
+
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+    file_parameters = {'status_id': status_id, 'last_update': last_update,
+                       'completion_time': completion_time, 'error_message': error_message,
+                       'job_id': job_id, 'granule_id': granule_id, 'filename': filename}
+    job_parameters = {'job_id': job_id, 'granule_id': granule_id}
+    engine = shared_db.get_user_connection(db_connect_info)
+    with engine.begin() as connection:
+        connection.execute(update_file_sql(), file_parameters)
```

Comment:
Possibly add some log statements here either as info or debug to help narrow down issues if an update fails.